src/terminal/parser/stateMachine.cpp

```diff
@@ -320,7 +299,7 @@ static constexpr bool _isOscInvalid(const wchar_t wch) noexcept
 // - True if it is. False if it isn't.
 static constexpr bool _isOscTerminator(const wchar_t wch) noexcept
 {
-    return wch == AsciiChars::BEL || _isStringTerminator(wch); // Bell character or C1 terminator
+    return wch == AsciiChars::BEL; // Bell character

```

Comment:
Should `BEL` also terminate DCS and SOS/PM/APC ?

Reply:
I don't think so. My understanding is that it's just supported for legacy reasons in OSC, but is not actually part of the DEC or ANSI standards. Not all terminals handled this the same way, but I think the majority (including XTerm) will only recognise BEL as a terminator in OSC, so I'd recommend we follow that behaviour.